src/main/java/org/folio/edge/sip2/repositories/LoginRepository.java

```diff
@@ -43,31 +39,32 @@ public Future<LoginResponse> login(Login login, SessionData sessionData) {
     final String user = login.getLoginUserId();
     final String password = login.getLoginPassword();
     final String locationCode = login.getLocationCode();
+    sessionData.setUsername(user);
+    sessionData.setPassword(password);
+
+    Future<String> authToken = null;
+    authToken = resourceProvider.loginWithSupplier(user,
+        () -> Future.succeededFuture(password), sessionData);
+
+    if (authToken == null) {
+      // Can't continue without an auth token
+      log.error("Login does not have a valid authentication token");
+      sessionData.setAuthenticationToken(null);
+      return Future.succeededFuture(LoginResponse.builder().ok(FALSE).build());
+    }
 
-    final JsonObject credentials = new JsonObject()
-        .put("username", user)
-        .put("password", password);
+    return authToken
+     .compose(token -> {
```

Comment:
Think you may need another space (or 3 spaces) before the block with the `.compose`.

Reply:
CheckStyle plugin does not agree to this :) the build fails if I increase/decrease the space count unfortunately.